pkg/snapstore/ocs_s3_snapstore.go

```diff
@@ -64,14 +210,46 @@ func ocsAuthOptionsFromEnv() (s3AuthOptions, error) {
 		insecureSkipVerify = ocsDefaultInsecureSkipVerify
 	}
 
-	ao := s3AuthOptions{
-		endpoint:           endpoint,
-		region:             region,
-		disableSSL:         disableSSL,
-		insecureSkipVerify: insecureSkipVerify,
-		accessKeyID:        accessKeyID,
-		secretAccessKey:    secretAccessKey,
+	ao := ocsAuthOptions{
+		Endpoint:           endpoint,
+		Region:             region,
+		DisableSSL:         disableSSL,
+		InsecureSkipVerify: insecureSkipVerify,
+		AccessKeyID:        accessKeyID,
+		SecretAccessKey:    secretAccessKey,
 	}
 
-	return ao, nil
+	return &ao, nil
+}
+
+func ocsAuthOptionsToGenericS3(options ocsAuthOptions) s3AuthOptions {
+	return s3AuthOptions{
+		endpoint:           options.Endpoint,
+		region:             options.Region,
+		accessKeyID:        options.AccessKeyID,
+		secretAccessKey:    options.SecretAccessKey,
+		disableSSL:         options.DisableSSL,
+		insecureSkipVerify: options.InsecureSkipVerify,
+	}
+}
+
+func isOCSConfigEmpty(config ocsAuthOptions) error {
+	if len(config.AccessKeyID) != 0 && len(config.Region) != 0 && len(config.SecretAccessKey) != 0 && len(config.Endpoint) != 0 {
+		return nil
+	}
+	return fmt.Errorf("ocs s3 credentials: region, secretAccessKey, endpoint or accessKeyID is missing")
+}
+
+func OCSSnapStoreHash(config *brtypes.SnapstoreConfig) (string, error) {
+	ao, err := getOCSAuthOptions("")
+	if err != nil {
+		return "", err
+	}
+
+	return getOCSHash(ao), nil
+}
+
+func getOCSHash(config *ocsAuthOptions) string {
+	data := fmt.Sprintf("%s%s%s%s%v%v", config.AccessKeyID, config.Endpoint, config.Region, config.SecretAccessKey, config.DisableSSL, config.InsecureSkipVerify)
```

Comment:
`config.DisableSSL`, `config.InsecureSkipVerify` can we remove these 2 as they are optional ?

Reply:
You are right :)
I've removed them from the hashing function.